lua/telescope/builtin/internal.lua

```diff
@@ -45,6 +45,73 @@ internal.builtin = function(opts)
   }):find()
 end
 
+do
+  internal.menu = function(opts)
+    if opts.callback == nil then
+      vim.cmd [[echoerr 'There is no callback']]
+      return
+    end
+
+    pickers.new(opts, {
+      prompt_title = opts.title or 'Menu',
+      finder = finders.new_tree {
+        results = opts.t,
+      },
+      sorter = conf.generic_sorter(opts),
+      attach_mappings = function(prompt_bufnr)
+        actions._goto_file_selection:replace(function()
+          local entry = actions.get_selected_entry()
+          local value = entry.value
+          local key = entry.key
+          if value == nil then
+            -- it is a leaf
+            opts.callback(key)
+            actions.close(prompt_bufnr)
+          elseif type(value) == "table" then
+            -- it is a node
+            actions.close(prompt_bufnr)
+            -- recurse
+            internal.menu{t = value, callback = opts.callback}
+            -- sometimes does not start insert for some reason
+            vim.api.nvim_input('i')
```

Comment:
There is an actually workaround for this issue. There is a local function inside `lua/telescope/actions/init.lua` which is used for `actions.run_builtin` and is called `do_close`. We could make that one public like this: `actions._do_close` similar to `actions._goto_file_selection` so you can do `actions._do_close(prompt_bufnr, true)` rather the `actions.close` and `nvim_input`